src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -427,14 +428,15 @@ public override string ReadToEnd()
             CheckAsyncTaskInProgress();
 
             // Call ReadBuffer, then pull data out of charBuffer.
-            StringBuilder sb = new StringBuilder(_charLen - _charPos);
+            _sb?.Clear();
+            _sb ??= new StringBuilder(_charLen - _charPos);
```

Comment:
If the `_sb` is `null`, then it doesn't need to be cleared, just the new one created.
So this can written more efficient:
```c#
if (_sb is null)
{
    _sb = new StringBuilder(_charLen - _charPos);
}
else
{
    _sb.Clear();
}
```
Look at the code generated from C# compiler.

Reply:
Done